src/Cms/AppPlugins.php

```diff
@@ -713,17 +713,22 @@ public function nativeComponent(string $component): Closure|false
 	 */
 	public static function plugin(
 		string $name,
-		array $extends = null
+		array $extends = null,
+		array $info = [],
+		string|null $root = null
```

Comment:
I feel we should already introduce the separate `$version` argument here as well. It's true we need the general `$info` array anyway, but IMO every info prop Kirby needs should be provided with a typed named argument like Bastian wrote. With a separate argument, we can advertise the use of the named argument in the docs from the start.

Reply:
As I wrote, what's every info prop? Are we adding named arguments for every possible composer entry? Which is why Bastian suggested to still keep it simple with just the version named argument.

Reply:
Exactly, but it is only in the `Plugin` constructor, not the `App::plugin()` method at the moment. Therefore my suggestion to add it there as well.